api/ocm/plugin/README.md

```diff
@@ -0,0 +1,151 @@
+# The OCM Library Plugin Concept
+
+The ocm library supports a plugin mechanism to provide further variants
+for OCM and library extension points without the need of extending and recompiling
+the OCM CLI (and other applications using this library).
+
+This plugin concept is not part of the OCM specification, because it is
+a feature of this library implementation.
+
+The following extension points are supported:
+
+- Access methods
+- Uploaders
+- Downloaders
+- Actions
+- Value sets (for example for routing slip entries)
+- Config types
+- Value Merge Handler (for example for label values in delta transports)
+- CLI Commands (for OCM CLI)
+- (transfer handlers)
+- (signing tools)
+- (input types for component version composition)
+
+## Plugin Technology
+
+A plugin is a simple executable, which might be written in any program language.
+The plugin has to provide a set of CLI commands for every extension point it provides new variations for.
+
+The data transfer between the library and the plugin is done via
+
+- command options (for inbound information)
+- standard input (for potentially large inbound content (streaming))
```

Comment:
What if a plugin needs multiple data streams? in that case STDIN would not be feasible for both.